lib/ansible/utils/collection_loader/_collection_finder.py

```diff
@@ -233,7 +233,7 @@ def __init__(self, paths=None, scan_sys_paths=True):
 
             # ensure we always have ansible_collections
             if os.path.basename(p) == 'ansible_collections':
-                p = os.path.dirname(p)
+                p = os.path.dirname(os.path.realpath(p))
```

Comment:
```
seen = set()
if os.path.basename(p) - 'ansible_collections:
    p=os.path.dirname(p)
    r=os.path.realpath(p)
    if r in seen:
         continue
     seen.add(r)
```

Reply:
This is actually handled right below in `good_paths`.

Reply:
Hey sorry, I made my changes before I saw these comments. @bcoca, don't lines 238 and 239 effectively skip dupes?

Reply:
the point is to dedeup on real path, but not change the path shown to the user as that might not match configured paths and create confusion. If it makes sense to do both things in one step instead of 2 iterations, we should do that, but do not change the actual path the user is expecting to see.